src/journal/journalctl.c

```diff
@@ -2753,25 +2786,15 @@ static int run(int argc, char *argv[]) {
         };
 
         if (arg_follow) {
-                if (arg_lines != 0 || arg_since_set) {
-                        r = show(&c);
-                        if (r < 0)
-                                return r;
-
-                        fflush(stdout);
-                }
+                _cleanup_(sd_event_unrefp) sd_event *e = NULL;
 
-                for (;;) {
-                        r = wait_for_change(j, poll_fd);
-                        if (r < 0)
-                                return r;
+                assert(poll_fd >= 0);
 
-                        r = show(&c);
-                        if (r < 0)
-                                return r;
+                r = setup_event(&c, poll_fd, &e);
```

Comment:
> One minor(?) behavior change is that 'journalctl --follow' now exits with 0 on terminate e.g. with ctrl-C.

That's actually a regression. I'm not sure if anyone relies on this, but previously we were returning correct information but now we're suppressing it.

E.g. when I run this silly example in fish, the result looks strange:
```console
[0] $ tail -f /etc/fstab | build/journalctl -f | grep foobar
^C⏎
[SIGINT|0|SIGINT] $
```